### PR TITLE
Contract wrapper that allows local signatures

### PR DESCRIPTION
**Pull request by pcppcp**
- implements a wrapper for web3 contract that allows signing contract
  calls with a local private key
- can be removed once middleware signatures are merged to web3: https://github.com/ethereum/web3.py/pull/649

### CODE REVIEW

File: raiden_libs/contract.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+DEFAULT_RETRY_INTERVAL = 3
+GAS_PRICE = 20 * denoms.gwei
+GAS_LIMIT_POT = 21000
+GAS_LIMIT_CONTRACT = 130000
+
+
+def create_signed_transaction(
+        private_key: str,
+        web3: Web3,
+        to: str,
```
Comment (palango) [inline raiden_libs/contract.py]: We should add a typing module and make a type for addresses. Like https://github.com/raiden-network/raiden-pathfinding-service/blob/master/pathfinder/utils/types.py

File: raiden_libs/contract.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        value: int=0,
+        data=b'',
+        nonce_offset: int = 0,
+        gas_price: int = GAS_PRICE,
+        gas_limit: int = GAS_LIMIT_POT
+) -> str:
+    """
```
Comment (palango) [inline raiden_libs/contract.py]: One line

File: raiden_libs/contract.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+def create_transaction(
+        web3: Web3,
+        from_: str,
+        to: str,
+        data: bytes = b'',
+        nonce_offset: int = 0,
+        value: int = 0,
+        gas_price: int = GAS_PRICE,
+        gas_limit: int = GAS_LIMIT_POT
+) -> Transaction:
```
Comment (palango) [inline raiden_libs/contract.py]: docstring pls

File: raiden_libs/contract.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        private_key: str,
+        contract: Contract,
+        func_name: str,
+        args: List[Any],
+        value: int=0,
+        nonce_offset: int = 0,
+        gas_price: int = GAS_PRICE,
+        gas_limit: int = GAS_LIMIT_POT
+) -> str:
+    """
```
Comment (palango) [inline raiden_libs/contract.py]: One line

File: raiden_libs/private_contract.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
+from raiden_libs.contract import create_signed_transaction
+from web3.utils.abi import get_abi_output_types
+from eth_abi import decode_abi
+from eth_utils import decode_hex
+
+
+class Callable():
```
Comment (palango) [inline raiden_libs/private_contract.py]: docstring

File: raiden_libs/test/fixtures/contracts.py [hunk 6]
```diff
@@ -54,13 +54,14 @@ def token_network_contract(
...
 ):
+    network_id = int(secret_registry_contract.web3.version.network)
     return deploy_tester_contract(
         'TokenNetwork',
         {
             'Token': standard_token_contract.address.encode(),
             'SecretRegistry': secret_registry_contract.address.encode()
         },
-        [standard_token_contract.address, secret_registry_contract.address]
+        [standard_token_contract.address, secret_registry_contract.address, network_id]
```
Comment (palango) [inline raiden_libs/test/fixtures/contracts.py]: How is this additional parameter handled in `deploy_tester_contract`? There are no changes there...

File: raiden_libs/utils/signing.py [hunk 7]
```diff
@@ -101,3 +103,13 @@ def address_from_signature(sig: bytes, msg: bytes):
 
 def eth_verify(sig: bytes, msg: str) -> str:
     return address_from_signature(sig, keccak256(msg))
+
+
+def sign_transaction(tx: Transaction, privkey: str, network_id: int):
+    # Implementing EIP 155.
```
Comment (palango) [inline raiden_libs/utils/signing.py]: Make this a docstring. Also a test would be nice.

File: raiden_libs/test/fixtures/contracts.py (hunk 6 above)
Comment (pcppcp) [reply raiden_libs/test/fixtures/contracts.py]: these are constructor arguments, deploy will throw if they're incorrect

File: raiden_libs/contract.py (hunk 1 above)
Comment (pcppcp) [reply raiden_libs/contract.py]: easy peasy, but let's make this a separate issue & PR

File: raiden_libs/test/fixtures/contracts.py (hunk 6 above)
Comment (palango) [reply raiden_libs/test/fixtures/contracts.py]: But did this change? I don't  find changes in the contracts!?

Comment (pcppcp) [reply raiden_libs/test/fixtures/contracts.py]: The change is from 23 days ago.. But maybe no test is using this fixture so it didn't cause problems.

File: raiden_libs/utils/signing.py (hunk 7 above)
Comment (pcppcp) [reply raiden_libs/utils/signing.py]: test is already included... if signature doesn't work, local signing won't work :)

**Pull request merged**